theseus/theseus_layer.py

```diff
@@ -5,11 +5,17 @@
 
 from typing import Any, Dict, Optional, Tuple
 
+from functools import partial
+
 import torch
 import torch.nn as nn
+from torch.autograd.function import once_differentiable
 
+from theseus import Variable, GaussNewton
```

Comment:
Our convention is to use relative imports in lib files and theseus global imports in tests and examples.
`from theseus.core import Variable`
`from theseus.optimizer.nonlinear import GaussNewton`